marconi-chain-index/test-lib/Helpers.hs

```diff
@@ -84,16 +85,20 @@ readAs as path = do
   H.leftFailM . liftIO $ C.readFileTextEnvelope as path'
 
 -- | An empty transaction
-emptyTxBodyContent :: C.Lovelace -> C.ProtocolParameters -> C.TxBodyContent C.BuildTx C.AlonzoEra
-emptyTxBodyContent fee pparams = C.TxBodyContent
+emptyTxBodyContent
+    :: (C.TxValidityLowerBound era, C.TxValidityUpperBound era)
```

Comment:
Reducing the number of args to this function is best to reduce code because not all tests need to set validity interval. Imo only pparams is necessary.

Reply:
We can't for a multi era implementation of `emptyTxBodyContent`. The reason is because the upper bound in Shelley era is mandatory, so we can't put an empty upper bound when creating a Shelley tx. 

We can discuss further on how to extend this, but one solution would be to create era specific modules which specialize the functions in this module.

Reply:
Ok, because in the plutus e2e tests will never use Shelley Era so `C.validityNoUpperBoundSupportedInEra` is used by default.